onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -911,10 +911,12 @@ static Status CreateEpContextModel(const ExecutionProviders& execution_providers
 
   size_t ini_size_threshold = ep_context_gen_options.output_external_initializer_size_threshold;
   std::filesystem::path external_ini_path = ep_context_gen_options.output_external_initializers_file_path;
+  bool force_embed_external_ini = false;
   if (external_ini_path.empty()) {
-    // Set the threshold to the max so all initializers are forced into the Onnx file
+    // if no external ini file specified, set force_embed_external_ini to true to avoid intermedia file creation
+    // and force all initializers embed into the Onnx file
     ini_size_threshold = SIZE_MAX;
-    external_ini_path = "./model_ext_ini.bin";
+    force_embed_external_ini = true;
```

Comment:
Would it make sense to move this flag inside of the `ModelSavingOptions` struct? The `ModelSavingOptions` struct should probably have all settings related to saving models with initializers:

https://github.com/microsoft/onnxruntime/blob/9534ab41a04bca790001e21bd05ff942f52fe098/include/onnxruntime/core/graph/model_saving_options.h#L10-L42